Ubuntu 13.10 | cmake error | -- Configuring incomplete, errors occurred!

I tried to install HHVM on a fresh installation of Ubuntu 13.10 using the following guide [Building and installing HHVM on Ubuntu 13.10](https://github.com/facebook/hhvm/wiki/Building-and-installing-HHVM-on-Ubuntu-13.10)

After running the `cmake .` command I get the following errors:

``` bash
CMake Error at CMake/EXTFunctions.cmake:20 (list):
  list sub-command REMOVE_AT requires list to be present.
Call Stack (most recent call first):
  hphp/runtime/ext/imagick/config.cmake:6 (HHVM_EXTENSION)
  hphp/runtime/ext/CMakeLists.txt:24 (include)


CMake Error at CMake/EXTFunctions.cmake:8 (list):
  list sub-command REMOVE_AT requires list to be present.
Call Stack (most recent call first):
  hphp/runtime/ext/imagick/config.cmake:12 (HHVM_LINK_LIBRARIES)
  hphp/runtime/ext/CMakeLists.txt:24 (include)


-- Configuring incomplete, errors occurred!
```

Here is the full output

``` bash
bogardo@hhvm:~/dev/hhvm$ cmake .
-- The C compiler identification is GNU 4.8.1
-- The CXX compiler identification is GNU 4.8.1
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
Using HPHP_HOME == /home/bogardo/dev/hhvm
-- Performing Test IS_AARCH64
-- Performing Test IS_AARCH64 - Failed
-- CMAKE_PREFIX_PATH was missing, proceeding anyway
CMake Warning at CMake/HPHPSetup.cmake:59 (message):
  chrpath not found, rpath will not be stripped from installed binaries
Call Stack (most recent call first):
  hphp/CMakeLists.txt:18 (include)


-- Performing Test HAVE_GCC_47
-- Performing Test HAVE_GCC_47 - Success
-- Performing Test HAVE_GCC_48
-- Performing Test HAVE_GCC_48 - Success
-- Found LibDL: /usr/lib/x86_64-linux-gnu/libdl.so
-- Performing Test LIBDL_NEEDS_UNDERSCORE
-- Performing Test LIBDL_NEEDS_UNDERSCORE - Failed
-- Boost version: 1.53.0
-- Found the following Boost libraries:
--   system
--   program_options
--   filesystem
--   regex
-- Found LIBGLOG: /usr/lib/x86_64-linux-gnu/libglog.so
-- Could NOT find LIBINOTIFY (missing:  LIBINOTIFY_LIBRARY)
-- Could NOT find LIBICONV (missing:  LIBICONV_LIBRARY)
-- Performing Test LIBICONV_CONST
-- Performing Test LIBICONV_CONST - Failed
-- MySQL Include dir: /usr/include/mysql  library dir: /usr/lib/x86_64-linux-gnu
-- MySQL client libraries: mysqlclient_r
-- Found LIBMEMCACHED: /usr/lib/x86_64-linux-gnu/libmemcached.so
-- Found PCRE: /usr/lib/x86_64-linux-gnu/libpcre.so
-- Found libevent: /usr/lib/x86_64-linux-gnu/libevent.so
-- Looking for evhttp_bind_socket_with_fd
-- Looking for evhttp_bind_socket_with_fd - not found
-- Could NOT find LibUODBC (missing:  LIBODBC_LIBRARIES LIBODBC_INCLUDE_DIRS)
-- Found LibJpeg: /usr/lib/x86_64-linux-gnu/libjpeg.so
-- Found LibPng: /usr/lib/x86_64-linux-gnu/libpng.so
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.26")
-- Found Freetype: /usr/lib/x86_64-linux-gnu/libfreetype.so
-- Found CURL: /usr/lib/x86_64-linux-gnu/libcurl.so (found version "7.32.0")
-- Looking for curl_multi_select
-- Looking for curl_multi_select - not found
-- Looking for curl_multi_wait
-- Looking for curl_multi_wait - found
-- Found LibXml2: /usr/lib/x86_64-linux-gnu/libxml2.so (found version "2.9.1")
-- Found LibXslt: /usr/lib/x86_64-linux-gnu/libxslt.so (found version "1.1.28")
-- Found EXPAT: /usr/lib/x86_64-linux-gnu/libexpat.so (found version "2.1.0")
-- Found ICU header files in /usr/include/x86_64-linux-gnu
-- Found ICU libraries: /usr/lib/x86_64-linux-gnu/libicuuc.so
-- Performing Test JEMALLOC_VERSION_3
-- Performing Test JEMALLOC_VERSION_3 - Success
-- Found jemalloc: /usr/lib/x86_64-linux-gnu/libjemalloc.so
-- Found Intel TBB
-- Found mcrypt: /usr/lib/libmcrypt.so
-- Found OpenSSL: /usr/lib/x86_64-linux-gnu/libssl.so;/usr/lib/x86_64-linux-gnu/libcrypto.so (found version "1.0.1e")
-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version "1.2.8")
-- Found BZip2: /usr/lib/x86_64-linux-gnu/libbz2.so (found version "1.0.6")
-- Looking for BZ2_bzCompressInit in /usr/lib/x86_64-linux-gnu/libbz2.so
-- Looking for BZ2_bzCompressInit in /usr/lib/x86_64-linux-gnu/libbz2.so - found
-- Found ONIGURUMA: /usr/lib/libonig.so
-- Found ldap: /usr/lib/x86_64-linux-gnu/libldap.so;/usr/lib/x86_64-linux-gnu/liblber.so
-- Found LIBPTHREAD: /usr/lib/x86_64-linux-gnu/libpthread.so
-- Found Readline: /usr/include
-- Could NOT find Editline (missing:  EDITLINE_LIBRARIES EDITLINE_INCLUDE_DIRS)
-- Found LibElf: /usr/lib/x86_64-linux-gnu/libelf.so
-- Performing Test ELF_GETSHDRSTRNDX
-- Performing Test ELF_GETSHDRSTRNDX - Success
-- Found LibDwarf: /usr/lib/libdwarf.a
-- Looking for dwarf_encode_leb128
-- Looking for dwarf_encode_leb128 - not found
Generating Release build
-- Performing Test FOLLY_IFUNC
-- Performing Test FOLLY_IFUNC - Failed
-- Looking for clock_gettime
-- Looking for clock_gettime - found
-- Looking for arpa/inet.h
-- Looking for arpa/inet.h - found
-- Looking for netinet/in.h
-- Looking for netinet/in.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for string.h
-- Looking for string.h - found
-- Looking for sys/socket.h
-- Looking for sys/socket.h - found
-- Looking for sys/time.h
-- Looking for sys/time.h - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of size_t
-- Check size of size_t - done
-- Check size of ssize_t
-- Check size of ssize_t - done
-- Check size of uint32_t
-- Check size of uint32_t - done
-- Check size of uint8_t
-- Check size of uint8_t - done
-- Looking for AF_LOCAL
-- Looking for AF_LOCAL - found
-- Looking for PF_LOCAL
-- Looking for PF_LOCAL - found
-- Looking for memset
-- Looking for memset - found
-- Looking for socket
-- Looking for socket - found
-- Looking for strerror
-- Looking for strerror - found
-- Found libevent: /usr/lib/x86_64-linux-gnu/libevent.so
-- Looking for event_get_version_number
-- Looking for event_get_version_number - found
-- Performing Test HAVE_LIBEVENT_209
-- Performing Test HAVE_LIBEVENT_209 - Success
-- Found libevent 2.0.9+
-- Looking for include file stdlib.h
-- Looking for include file stdlib.h - found
-- Looking for include file assert.h
-- Looking for include file assert.h - found
-- Looking for strcasecmp
-- Looking for strcasecmp - found
-- Looking for strchr
-- Looking for strchr - found
-- Looking for include file sys/stat.h
-- Looking for include file sys/stat.h - found
-- Looking for include file stdlib.h
-- Looking for include file stdlib.h - found
-- Looking for include file string.h
-- Looking for include file string.h - found
-- Looking for include file memory.h
-- Looking for include file memory.h - found
-- Looking for include file strings.h
-- Looking for include file strings.h - found
-- Looking for include file inttypes.h
-- Looking for include file inttypes.h - found
-- Looking for include file unistd.h
-- Looking for include file unistd.h - found
-- Looking for include file dlfcn.h
-- Looking for include file dlfcn.h - found
-- Looking for fdatasync
-- Looking for fdatasync - found
-- Looking for usleep
-- Looking for usleep - found
-- Looking for fullfsync
-- Looking for fullfsync - not found
-- Looking for localtime_r
-- Looking for localtime_r - found
-- Looking for gmtime_r
-- Looking for gmtime_r - found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Check if the system is big endian
-- Searching 16 bit integer
-- Check size of unsigned short
-- Check size of unsigned short - done
-- Using unsigned short
-- Check if the system is big endian - little endian
-- Looking for include file locale.h
-- Looking for include file locale.h - found
-- Looking for include file limits.h
-- Looking for include file limits.h - found
-- Check size of int32_t
-- Check size of int32_t - done
-- Looking for strtoll
-- Looking for strtoll - found
-- Looking for atoll
-- Looking for atoll - found
-- Looking for strftime
-- Looking for strftime - found
-- Looking for setlocale
-- Looking for setlocale - found
-- Boost version: 1.53.0
-- Found the following Boost libraries:
--   system
--   program_options
--   filesystem
--   regex
-- Looking for _close
-- Looking for _close - not found
-- Looking for _dup
-- Looking for _dup - not found
-- Looking for _fdopen
-- Looking for _fdopen - not found
-- Looking for _fileno
-- Looking for _fileno - not found
-- Looking for _open
-- Looking for _open - not found
-- Looking for _snprintf
-- Looking for _snprintf - not found
-- Looking for _strdup
-- Looking for _strdup - not found
-- Looking for _stricmp
-- Looking for _stricmp - not found
-- Looking for fseeko
-- Looking for fseeko - found
-- Looking for ftello
-- Looking for ftello - found
-- Looking for open
-- Looking for open - found
-- Looking for mkstemp
-- Looking for mkstemp - found
-- Looking for snprintf
-- Looking for snprintf - found
-- Looking for strdup
-- Looking for strdup - found
-- Looking for MoveFileExA
-- Looking for MoveFileExA - not found
-- Looking for include file sys/types.h
-- Looking for include file sys/types.h - found
-- Looking for include file inttypes.h
-- Looking for include file inttypes.h - found
-- Looking for include file stdint.h
-- Looking for include file stdint.h - found
-- Check size of __int8
-- Check size of __int8 - failed
-- Check size of int8_t
-- Check size of int8_t - done
-- Check size of uint8_t
-- Check size of uint8_t - done
-- Check size of __int16
-- Check size of __int16 - failed
-- Check size of int16_t
-- Check size of int16_t - done
-- Check size of uint16_t
-- Check size of uint16_t - done
-- Check size of __int32
-- Check size of __int32 - failed
-- Check size of int32_t
-- Check size of int32_t - done
-- Check size of uint32_t
-- Check size of uint32_t - done
-- Check size of __int64
-- Check size of __int64 - failed
-- Check size of int64_t
-- Check size of int64_t - done
-- Check size of uint64_t
-- Check size of uint64_t - done
-- Check size of short
-- Check size of short - done
-- Check size of int
-- Check size of int - done
-- Check size of long
-- Check size of long - done
-- Check size of long long
-- Check size of long long - done
-- Check size of off_t
-- Check size of off_t - done
-- Check size of size_t
-- Check size of size_t - done
-- Check size of ssize_t
-- Check size of ssize_t - done
-- Boost version: 1.53.0
-- Found the following Boost libraries:
--   thread
--   system
-- Boost version: 1.53.0
-- Found the following Boost libraries:
--   thread
--   system
-- Found ocamlc: /usr/bin/ocamlc
-- Found ocamlc.opt: /usr/bin/ocamlc.opt
-- Found LibMagicWand: /usr/lib/x86_64-linux-gnu/libMagickWand.so
CMake Error at CMake/EXTFunctions.cmake:20 (list):
  list sub-command REMOVE_AT requires list to be present.
Call Stack (most recent call first):
  hphp/runtime/ext/imagick/config.cmake:6 (HHVM_EXTENSION)
  hphp/runtime/ext/CMakeLists.txt:24 (include)


CMake Error at CMake/EXTFunctions.cmake:8 (list):
  list sub-command REMOVE_AT requires list to be present.
Call Stack (most recent call first):
  hphp/runtime/ext/imagick/config.cmake:12 (HHVM_LINK_LIBRARIES)
  hphp/runtime/ext/CMakeLists.txt:24 (include)


-- Configuring incomplete, errors occurred!
bogardo@hhvm:~/dev/hhvm$
```

--- a/CMake/EXTFunctions.cmake
+++ b/CMake/EXTFunctions.cmake
@@ -5,20 +5,19 @@
 # will differ slightly.
 
 macro(HHVM_LINK_LIBRARIES EXTNAME)
-	list(REMOVE_AT ARGV 0)
-	foreach (lib ${ARGV})
+	foreach (lib ${ARGN})
 		list(APPEND HRE_LIBRARIES ${lib})
 	endforeach()
 endmacro()
 
 function(HHVM_ADD_INCLUDES EXTNAME)
-	list(REMOVE_AT ARGV 0)
-	include_directories(${ARGV})
+    include_directories(${ARGN})
 endfunction()
 
 macro(HHVM_EXTENSION EXTNAME)
-	list(REMOVE_AT ARGV 0)
-	list(APPEND CXX_SOURCES "${HRE_CURRENT_EXT_PATH}/${ARGV}")
+	foreach (src ${ARGN})
+	    list(APPEND CXX_SOURCES "${HRE_CURRENT_EXT_PATH}/${src}")
+    endforeach()
 endmacro()
 
 function(HHVM_SYSTEMLIB EXTNAME SOURCE_FILE)

--- a/CMake/HPHPIZEFunctions.cmake
+++ b/CMake/HPHPIZEFunctions.cmake
@@ -5,25 +5,21 @@
 # will differ slightly.
 
 function(HHVM_LINK_LIBRARIES EXTNAME)
-	list(REMOVE_AT ARGV 0)
-	target_link_libraries(${EXTNAME} ${ARGV})
+	target_link_libraries(${EXTNAME} ${ARGN})
 endfunction()
 
 function(HHVM_ADD_INCLUDES EXTNAME)
-	list(REMOVE_AT ARGV 0)
-	include_directories(${ARGV})
+	include_directories(${ARGN})
 endfunction()
 
 function(HHVM_EXTENSION EXTNAME)
-	list(REMOVE_AT ARGV 0)
-	add_library(${EXTNAME} SHARED ${ARGV})
+	add_library(${EXTNAME} SHARED ${ARGN})
 	set_target_properties(${EXTNAME} PROPERTIES PREFIX "")
 	set_target_properties(${EXTNAME} PROPERTIES SUFFIX ".so")
 endfunction()
 
 function(HHVM_SYSTEMLIB EXTNAME)
-	list(REMOVE_AT ARGV 0)
-	foreach (SLIB ${ARGV})
+	foreach (SLIB ${ARGN})
 		embed_systemlib_byname(${EXTNAME} ${SLIB})
 	endforeach()
 	embed_systemlibs(${EXTNAME} "${EXTNAME}.so")

